### PR TITLE
openthread: upmerge to commit `gaad6518ee`

### PR DESCRIPTION
**Pull request by edmont**
- Update manifest to include cherry-picked Zephyr commits which include OpenThread upmerge to commit `gaad6518ee` and required additions by the upmerge.
- Add necessary changes to OT samples after the upmerge.
- Update manifest to include new OT libraries in nrfxlib created out of the previous changes.

Note: `MBEDTLS_ECDSA_DETERMINISTIC` not supported yet.

### CODE REVIEW

File: doc/nrf/ug_thread_ot_memory.rst [hunk 1]
```diff
@@ -77,25 +76,44 @@ The following tables present memory requirements for samples running on the :ref
...
-   | :ref:`CLI <ot_cli_sample_minimal>` | mtd       |                               298 |                             0 |                        24 |             190 |                                79 |                             0 |              49 |
-   +------------------------------------+-----------+-----------------------------------+-------------------------------+---------------------------+-----------------+-----------------------------------+-------------------------------+-----------------+
+   +------------------------------------+-----------+---------------------------+----------------------+---------------------------+-----------------+---------------------------+----------------------+-----------------+
+   | Sample                             | Variant   |   ROM OT stack + App [kB] |   ROM BLE stack [kB] |   Persistent storage [kB] |   Free ROM [kB] |   RAM OT stack + App [kB] |   RAM BLE stack [kB] |   Free RAM [kB] |
```
Comment (ru-fu) [inline doc/nrf/ug_thread_ot_memory.rst:80]: We cannot use BLE in the docs.
You could try forcing line breaks in the column headers with a line block - I know this works in list tables, not sure if you can do it in ascii tables as well ...
In list tables you would use it like this:
```
   * - Line-break example
     - x
     - | First line
       | Second line
```

File: doc/nrf/ug_thread_ot_memory.rst [hunk 2]
```diff
@@ -77,25 +76,44 @@ The following tables present memory requirements for samples running on the :ref
...
+   +------------------------------------+-----------+---------------------------+----------------------+---------------------------+-----------------+---------------------------+----------------------+-----------------+
+   | :ref:`CLI <ot_cli_sample_minimal>` | mtd       |                       313 |                   63 |                        24 |             112 |                        80 |                   10 |              38 |
+   +------------------------------------+-----------+---------------------------+----------------------+---------------------------+-----------------+---------------------------+----------------------+-----------------+
+
+.. _thread_ot_memory_5340:
+
+nRF5340 DK RAM and flash memory requirements
+*********************************************
+
+The following tables present memory requirements for samples running on the :ref:`nRF5340 DK <gs_programming_board_names>` (:ref:`nrf5340dk_nrf5340 <zephyr:nrf5340dk_nrf5340>`) with the hardware cryptography support provided by the CC312.
```
Comment (ru-fu) [inline doc/nrf/ug_thread_ot_memory.rst:107]: Let's add a link to CC312:
```suggestion
The following tables present memory requirements for samples running on the :ref:`nRF5340 DK <gs_programming_board_names>` (:ref:`nrf5340dk_nrf5340 <zephyr:nrf5340dk_nrf5340>`) with the hardware cryptography support provided by the :ref:`CC312 <nrfxlib:nrf_cc3xx_platform_readme>`.
```
If https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/crypto/doc/nrf_cc310_platform.html is the correct link?

**Review by ru-fu — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_thread_ot_memory.rst (hunk 1 above)
Comment (edmont) [reply doc/nrf/ug_thread_ot_memory.rst:80]: The table is generated automatically by `python-tabulate`. By the moment I'm okay with going back to Bluetooth LE.

File: doc/nrf/ug_thread_ot_memory.rst (hunk 2 above)
Comment (edmont) [reply doc/nrf/ug_thread_ot_memory.rst:107]: It should be. Shall we add a link for CC310 above as well?

Comment (ru-fu) [reply doc/nrf/ug_thread_ot_memory.rst:107]: Yes, please - should be the same reference, just different text.

**Review by MarekPorwisz — APPROVED** [review]

**Review by ru-fu — APPROVED** [review]

File: west.yml [hunk 3]
```diff
@@ -50,7 +50,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 8d7fe0bf3017968945d462175bbb780f7a7119ca
+      revision: pull/440/head
```
Comment (rlubos) [inline west.yml:53]: Please update SHA: `009b065f52ae8a07ab3265b022667c525fa62a9a`

File: west.yml [hunk 4]
```diff
@@ -99,7 +99,7 @@ manifest:
     - name: nrfxlib
       repo-path: sdk-nrfxlib
       path: nrfxlib
-      revision: 560a0fd8f2cac3a33d1e667279877bd053e02257
+      revision: pull/388/head
```
Comment (rlubos) [inline west.yml:102]: Please update SHA: `9017da739a02e51a3e152df8d2efc62bb4111b2f`

**Review by rlubos — APPROVED** [review]

**Pull request merged**